### PR TITLE
VITIS-13441 : Allow shim to read a special flag from xrt.ini for dummy app

### PR DESCRIPTION
**Pull request by aktondak**
#### Problem solved by the commit
https://jira.xilinx.com/browse/VITIS-13441 : Read "special" flag from xrt.ini for dummy app

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
[Read "special" flag from xrt.ini for dummy app](https://jira.xilinx.com/browse/VITIS-13441)
Special request from HP

#### How problem was solved, alternative solutions (if any) and why they were rejected
Problem was solved by introducing a new parameter in xrt.ini which can be queried by shim

#### Risks (if any) associated the changes in the commit
None

#### What has been tested and how, request additional testing if necessary
**Scenario 1** tested by Programmatically setting ini parameter through xrt::ini API and querying within the host app.
**Scenario 2** tested via the xrt_flow (https://gitenterprise.xilinx.com/XRT/testcases/tree/master/xrt_flow) app by giving a dummy xrt.ini in the directory and printing out the parameter value at run-time :
**Test Scenario 1 (Setting through xrt.ini):**
```
Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt_flow_test_pause.exe -x Dec23_CE1_CD1_MT1_1x4_c0d7b_opt_config2_w8_c0_dummy.xclbin -w mem_to_core_transfer\dpu_seq -p -s 1
Open ddr init successfully!
Host test code start...
Host test code is creating device object...
Host test code is loading xclbin object...Dec23_CE1_CD1_MT1_1x4_c0d7b_opt_config2_w8_c0_dummy.xclbin
Host test code found kernel: DPU_1x4
Host code is registering xclbin to the device...
Host code is creating hw_context...
**Runtime.dummy_app_context_type value is proxy**
```
xrt.ini:
```
[Debug]
 debug = true
 profile = false
 
[Runtime]
 dummy_app_context_type = proxy
```
**Test Scenario 2 (Setting programmatically through host app):**
Added the set API to df-bw testcase within xrt-smi to test correct setting of 'ini' parameter at run-time
```
xrt::ini::set("Runtime.dummy_app_context_type", "proxy"); 
logger(ptree, " Runtime.dummy_app_context_type", boost::str(boost::format("The value of Runtime.dummy_app_context_type is '%s'") % xrt_core::config::get_dummy_app_context_type()));
```
```
 Runtime.dummy_app_context_type                : The value of Runtime.dummy_app_context_type is 'proxy'
```
#### Documentation impact (if any)
None

### CODE REVIEW

**Review by stsoe — COMMENTED** [review]
Isn't the ask that the host application should be able to write the value for the config variable?  Did you test this?

While the PR description refers to a Jira ticket, it is still not clear exactly what the purpose of this dummy config option is?  Is it not possible to give the config variable a more descriptive name?

**Comment by aktondak** [comment]
> Isn't the ask that the host application should be able to write the value for the config variable? Did you test this?
> 
> While the PR description refers to a Jira ticket, it is still not clear exactly what the purpose of this dummy config option is? Is it not possible to give the config variable a more descriptive name?

I have tested through setting ini parameter through "xrt::ini::set". I have updated the Description with test scenario 2 for it.
The parameter name is derived from the proposal PDF which is a part of umbrella CR : https://jira.xilinx.com/browse/VITIS-13440. Open to other names as well. 
Referring to @pranjalv-xilinx for driver level purpose of this config option.

File: src/runtime_src/core/common/config_reader.h [hunk 1]
```diff
@@ -711,6 +711,13 @@ get_force_program_xclbin()
   return value;
 }
 
+inline std::string
+get_dummy_app_context_type()
```
Comment (pranjalv-xilinx) [inline src/runtime_src/core/common/config_reader.h:715]: can we remove "dummy" from the name and instead just call it "context_type"

Comment (aktondak) [reply src/runtime_src/core/common/config_reader.h:715]: Should I rename this to "hardware_context_type" ? @stsoe mentioned the parameter name should be more descriptive.

Comment (stsoe) [reply src/runtime_src/core/common/config_reader.h:715]: > Should I rename this to "hardware_context_type" ? @stsoe mentioned the parameter name should be more descriptive.

Yes, that would be better.

**Pull request merged**